### PR TITLE
Mean revenue by country

### PR DESCRIPTION
**Pull request by hnsun36**
Calculate the mean revenue by different countries.

### CODE REVIEW

File: explore_utils.py [hunk 1]
```diff
@@ -101,3 +101,22 @@ def find_channel_grouping_revenue(dataset):
...
+    
+    returns:
+       Tuple (dict, dict) containing mapping from geoNetwork.country to count
+       and mapping from geoNetwork.country to average revenue in dollars.
+    """
+    train_df = dataset.train
+    df = pd.DataFrame(train_df, columns=['geoNetwork.country', 'totals.transactionRevenue', 'fullVisitorId'])
+    df['totals.transactionRevenue'] = df['totals.transactionRevenue'].fillna(0)
+    groupings = df['geoNetwork.country'].unique()
+    counts = {grouping: df[df['geoNetwork.country'] == grouping].shape[0] for grouping in groupings}
```
Comment (jonathancstroud) [inline explore_utils.py]: (Optional) This can be done much more cleanly using pd.groupby.

File: explore.py [hunk 2]
```diff
@@ -46,6 +46,16 @@ def main(args):
     print('Mean Total Revenue by Channel Grouping')
     for key in sorted(means, key=means.get, reverse=True):
         print('\t{}: ${:.2f}'.format(key, means[key]))
+        
+    # geoNetwork.country and customer revenue (#31)
+    counts, means = explore_utils.find_country_revenue(data)
+    print('Country Counts:')
+    for key in sorted(counts, key=counts.get, reverse=True):
```
Comment (jonathancstroud) [inline explore.py]: Please only print the top 5 or 10 countries. Currently this prints hundreds of lines.

File: explore.py [hunk 3]
```diff
@@ -46,6 +46,16 @@ def main(args):
...
+        print('\t{}: {}'.format(key, counts[key]))
+
+    print('Mean Total Revenue by Country')
+    for key in sorted(means, key=means.get, reverse=True):
```
Comment (jonathancstroud) [inline explore.py]: Same as above.

File: explore_utils.py (hunk 1 above)
Comment (jonathancstroud) [inline explore_utils.py]: Does this ignore missing countries? Might be good to know how much revenue is associated with no country.

File: explore.py [hunk 4]
```diff
@@ -46,6 +46,16 @@ def main(args):
     print('Mean Total Revenue by Channel Grouping')
     for key in sorted(means, key=means.get, reverse=True):
         print('\t{}: ${:.2f}'.format(key, means[key]))
+
```
Comment (naitian) [inline explore.py]: Optional, but please remove trailing whitespace from blank lines.

File: explore_utils.py [hunk 5]
```diff
@@ -101,3 +101,22 @@ def find_channel_grouping_revenue(dataset):
...
+        dataset (Dataset): the google analystics dataset
+    
+    returns:
+       Tuple (dict, dict) containing mapping from geoNetwork.country to count
+       and mapping from geoNetwork.country to average revenue in dollars.
+    """
+    train_df = dataset.train
+    df = pd.DataFrame(train_df, columns=['geoNetwork.country', 'totals.transactionRevenue', 'fullVisitorId'])
+    df['totals.transactionRevenue'] = df['totals.transactionRevenue'].fillna(0)
+    groupings = df['geoNetwork.country'].unique()
```
Comment (naitian) [inline explore_utils.py]: Change the `groupings` variable name to something more descriptive, like `countries`.

**Pull request closed without merging**